### ISSUE TITLE
Mocktail test version out of date and blocking libraries

### ISSUE THREAD

**Issue by JustinmClapperton** [felangel/mocktail#178]
**Describe the bug**
Im trying to update another framework we use and mocktail (I see it hasnt been updated in 10+ months) is using an old version of `test` where newer libraries rely on newer versions.

Is there a plan to update Mocktail in the near future or has it been abandoned? or should I learn/convert to Mockito?

Thanks for the input.

```
Because test >=1.16.0-nullsafety.19 <1.16.6 depends on test_api 0.2.19 and test >=1.16.6 <1.17.10 depends on analyzer ^1.0.0, test >=1.16.0-nullsafety.19 <1.17.10 requires test_api 0.2.19 or analyzer ^1.0.0.
And because test >=1.17.10 <1.20.0 depends on analyzer >=1.0.0 <3.0.0 and test >=1.20.0 <1.21.2 depends on test_api 0.4.9, test >=1.16.0-nullsafety.19 <1.21.2 requires test_api 0.2.19 or 0.4.9 or analyzer >=1.0.0 <3.0.0.
And because test >=1.21.0 <1.21.6 depends on analyzer >=2.0.0 <5.0.0 and test >=1.21.6 <1.21.7 depends on test_api 0.4.14, test >=1.16.0-nullsafety.19 <1.21.7 requires test_api 0.2.19 or 0.4.9 or 0.4.14 or analyzer >=1.0.0 <5.0.0.
And because test >=1.21.7 <1.22.0 depends on test_api 0.4.15 and test >=1.22.0 <1.22.1 depends on test_api 0.4.16, test >=1.16.0-nullsafety.19 <1.22.1 requires test_api 0.2.19 or 0.4.9 or 0.4.14 or 0.4.15 or 0.4.16 or analyzer >=1.0.0 <5.0.0.
And because test >=1.22.1 <1.22.2 depends on test_api 0.4.17 and test >=1.22.2 depends on test_api 0.4.18, test >=1.16.0-nullsafety.19 requires test_api 0.2.19 or 0.4.9 or 0.4.14 or 0.4.15 or 0.4.16 or 0.4.17 or 0.4.18 or analyzer >=1.0.0 <5.0.0.
Because no versions of drift_dev match >2.4.1 <3.0.0 and drift_dev 2.4.1 depends on analyzer ^5.2.0, drift_dev ^2.4.1 requires analyzer ^5.2.0.
Thus, if drift_dev ^2.4.1 and test >=1.16.0-nullsafety.19 then test_api 0.2.19 or 0.4.9 or 0.4.14 or 0.4.15 or 0.4.16 or 0.4.17 or 0.4.18.
And because mocktail >=0.1.3 depends on test ^1.16.0, if drift_dev ^2.4.1 and mocktail >=0.1.3 then test_api 0.2.19 or 0.4.9 or 0.4.14 or 0.4.15 or 0.4.16 or 0.4.17 or 0.4.18.
And because app depends on flutter_test from sdk which depends on test_api 0.4.12, drift_dev ^2.4.1 is incompatible with mocktail >=0.1.3.
So, because routific_driverapp depends on both drift_dev ^2.4.1 and mocktail ^0.3.0, version solving failed.
pub get failed (1; So, because app depends on both drift_dev ^2.4.1 and mocktail ^0.3.0, version solving failed.)
```

**Comment by felangel** [comment]
Hi @JustinmClapperton 👋 
Thanks for opening an issue!

The version of test being used by mocktail supports the latest version of package:test. This issue is a duplicate of https://github.com/felangel/mocktail/issues/164 and has been fixed in the master channel of Flutter. Bumping the version of package:test in mocktail will not fix this issue as the constraint in mocktail is very wide as supports the latest version of package:test.

**Comment by JustinmClapperton** [comment]
hmm, the only way I can resolve this issue is removing mocktail.  All other packages work fine.  So im not sure how to resolve at this point then.   The library in conflict right now is Drift.  

I will look at the duplicate thread and maybe find an answer there but at a brief glance it looked like it had no conclusion either.

**Comment by felangel** [comment]
> hmm, the only way I can resolve this issue is removing mocktail. All other packages work fine. So im not sure how to resolve at this point then. The library in conflict right now is Drift.
> 
> I will look at the duplicate thread and maybe find an answer there but at a brief glance it looked like it had no conclusion either.

You can either upgrade to the master channel of Flutter temporarily or add a dependency_override for `package:test_api` as a temporary fix:

```yaml
dependency_overrides:
  test_api: 0.4.16
```

**Comment by amondnet** [comment]
@JustinmClapperton @felangel  #180

**Comment by felangel** [comment]
@amondnet this change should not be necessary. You should be able to fix the resolution issues by upgrading to the latest stable flutter channel (3.7).

**Comment by felangel** [comment]
Closing this since the issue is resolved on the new stable flutter channel (3.7)
